message.go

```diff
@@ -139,24 +138,18 @@ func (m *Message) WriteTo(w io.Writer) (int64, error) {
 	return total, nil
 }
 
-// DecodeMessage deseralizes data (as []byte) and creates a new Message
+// DecodeMessage deserializes data (as []byte) and creates a new Message
 func DecodeMessage(b []byte) (*Message, error) {
 	var msg Message
 
-	msg.Timestamp = int64(binary.BigEndian.Uint64(b[:8]))
-	msg.Attempts = binary.BigEndian.Uint16(b[8:10])
-
-	buf := bytes.NewBuffer(b[10:])
-
-	_, err := io.ReadFull(buf, msg.ID[:])
-	if err != nil {
-		return nil, err
+	if len(b) < 11+MsgIDLength {
```

Comment:
https://github.com/nsqio/nsq/blob/73cb4384556d670446da8d9a3fe0117d19b41f17/nsqd/message.go#L14

We're already doing something similar on the `nsqd` side so let's keep it consistent (which, btw, I'm realizing is also eligible for this same optimization).

Feel free to _also_ copy the message format into the function comment:

http://nsq.io/clients/tcp_protocol_spec.html#data-format

```
[x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x]...
|       (int64)        ||    ||      (hex string encoded in ASCII)           || (binary)
|       8-byte         ||    ||                 16-byte                      || N-byte
------------------------------------------------------------------------------------------...
  nanosecond timestamp    ^^                   message ID                       message body
                       (uint16)
                        2-byte
                       attempts
```